src/reflection.jl

```diff
@@ -73,6 +69,8 @@ function find_callsites(interp::CthulhuInterpreter, CI::Union{Core.CodeInfo, IRC
                             mici = MICallInfo(mi, rt)
                             return is_cached(mi) ? mici : UncachedCallInfo(mici)
                         end
+                    elseif isa(info, MethodResultPure)
+                        return Any[PureCallInfo(only(process_info(info.info)))]
```

Comment:
I agree with moving the handling of `MethodResultPure` into `process_info`, but I don't think `PureCallInfo` should be subtype of `WrappedCallInfo`.
`MethodResultPure` often doesn't wrap other call infos (e.g. see [here](https://github.com/JuliaLang/julia/blob/795935fd3a2d97b2f948cfb82a18da48743b622d/base/compiler/abstractinterpretation.jl#L1291)), and passing such cases can result in error:
```julia
julia> @eval descend(; optimize=false) do
           length($(Core.svec(0,1,2)))
       end
ERROR: ArgumentError: Collection is empty, must contain exactly 1 element
```
, and also it might be misleading to print "uncached" (i.e. when an user annotated a method as "pure" https://github.com/JuliaLang/julia/blob/795935fd3a2d97b2f948cfb82a18da48743b622d/base/compiler/abstractinterpretation.jl#L60) because `MethodResultPure` means super aggressive constant propagation and it's really opposite to "deoptimization", which "uncached" callsites usually mean.